src/Http/Routing/src/RouterMiddleware.cs

```diff
@@ -40,11 +40,16 @@ public async Task Invoke(HttpContext httpContext)
             }
             else
             {
-                httpContext.Features[typeof(IRoutingFeature)] = new RoutingFeature()
+                var routingFeature = new RoutingFeature()
                 {
-                    RouteData = context.RouteData,
+                    RouteData = context.RouteData
                 };
 
+                // Set the RouteValues on the current request
```

Comment:
If you want extra #RyanPoints update this comment to say that this is being down to keep `IRoutingFeature` in sync with `IRouteValuesFeature`. If you want even more extra points include ascii at of N'sync